src/pc/common/BitSet.js

```diff
@@ -0,0 +1,104 @@
+// we use 5 address bits for word since JS bit operations treat numbers as 32-bit integers
+const ADDRESS_BITS_PER_WORD = 5
+
+// Returns index of the word containing given bit
+function getWordIndex (bitIndex) {
+  return bitIndex >> ADDRESS_BITS_PER_WORD
+}
+
+// implementation of the bit set of variable length
+// heavily based on Java BitSet class
+class BitSet {
```

Comment:
This class is not needed, use BitArray instead, there is no need for duplicated code.